Embedded Software/TelemetryControlUnit/src/logger.cpp

```diff
@@ -0,0 +1,250 @@
+#include "logger.h"
+#include "debug.h"
+#include <Arduino.h>
+#include <TimeLib.h>
+#include <SD.h>
+
+#define MAX_BUFFERED_MESSAGES 50 // max number of buffered CAN messages before logging to SD card
+#define MAX_LOG_FILE_SIZE 100000000U // 100 Megabytes
+
+File logFile; // file logging object
+
+
+// Buffer information (use 2 buffers to prevent overwrites during logging delays)
+int buf1Length = 0;
+int buf2Length = 0;
+message_format_t messageBuf1[MAX_BUFFERED_MESSAGES]; 
+message_format_t messageBuf2[MAX_BUFFERED_MESSAGES]; 
+bool usingBuf1 = true; // true when using buf1, false when using buf2
+
+char fileName[19]; // format is log-1652888997.txt (19 digits as 10 for timestamp, 8 for text, 1 for termination)
+bool initialized = false;
+
+// Timing information
+uint32_t lastLogTime;
+uint32_t minLogFrequency;
+uint32_t startUpTimeMillis;
+uint32_t startUpTimeRTC;
+
+
+/**
+ * @brief Wrapper function to pass to time sync function
+ * 
+ * @return time_t 
+ */
+static time_t getTeensy3Time()
+{
+  return Teensy3Clock.get();
+}
+
+
+/**
+ * @brief Sets the logging file name using the format log-NUM.txt, where NUM is either the current 
+ *        unix time if using the RTC or a valid numeric index starting from 0.
+ * 
+ */
+static void setFileName() {
```

Comment:
Ngl I have a shitty understanding of like static vs const and their uses, but it feels like you're overusing static, but ngl I forget how to use them lol

Reply:
When using this C-style functional way of programming, `static` just limits the scope of the function/variable to be in the current file. So this `setFileName()` function can only be used in logger.cpp. It's not super necessary, just more of a convention (label the functions not defined in the header file as static so that anyone looking at the code knows it's a local function).